devicetypes/smartthings/zigbee-stateless-curtain-window-shade.src/zigbee-stateless-curtain-window-shade.groovy

```diff
@@ -0,0 +1,127 @@
+/**
+ *
+ *	Copyright 2019 SmartThings
+ *
+ *	Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except
+ *	in compliance with the License. You may obtain a copy of the License at:
+ *
+ *		http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *	Unless required by applicable law or agreed to in writing, software distributed under the License is distributed
+ *	on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License
+ *	for the specific language governing permissions and limitations under the License.
+ */
+import physicalgraph.zigbee.zcl.DataType
+import physicalgraph.zigbee.clusters.iaszone.ZoneStatus
+metadata {
+    definition(name: "Zigbee Stateless Curtain Window Shade", namespace: "smartthings", author: "SmartThings", ocfDeviceType: "oic.d.blind", mnmn: "SmartThings", vid: "generic-curtaion") {
+        capability "Actuator"
+        capability "Configuration"
+        capability "Refresh"
+        capability "Health Check"
+        capability "Switch Level"
+        capability "Stateless Curtain Power Button"
+        capability "Window Shade"
+
+        fingerprint profileId: "0104", inClusters: "0000, 0003, 0004, 0005, 0102", outClusters: "000A", manufacturer: "Feibit Co.Ltd", model: "FTB56-ZT218AK1.6", deviceJoinName: "Wistar Curtain Motor(CMJ)" // SY-IoT201-BD
+    }
+}
+
+private getCLUSTER_WINDOW_COVERING() { 0x0102 }
+private getATTRIBUTE_CURRENT_LEVEL() { 0x0000 }
+
+
+def parse(String description) {
+    log.debug "description:- ${description}"
+    def map = [:]
+    def resultMap = zigbee.getEvent(description)
+    log.debug "resultMap:- ${resultMap}"
+    if(resultMap){
+        map = resultMap;
+    }else{
+        Map descMap = zigbee.parseDescriptionAsMap(description)
+        log.debug "descMap:- ${descMap}"
+        if (descMap?.clusterInt == zigbee.LEVEL_CONTROL_CLUSTER && descMap.value) {
+            def valueInt = Math.round((zigbee.convertHexToInt(descMap.value)) / 255 * 100)
+            map = [name: "level", value: valueInt]
+        }
+    }
+    if(map?.name == "level"){
+        if(0 == map.value){
+            sendEvent(name: "windowShade", value: "closed")
+        }else if(100 == map.value){
+            sendEvent(name: "windowShade", value: "open")
+        }else{
+            sendEvent(name: "windowShade", value: "partially open")
+        }
+        log.debug "map:- ${map}"
+        sendEvent(map)
+    }
+}
+
+def close() {
+    log.info "close()"
+    sendEvent(name: "windowShade", value: "closing")
+    zigbee.command(CLUSTER_WINDOW_COVERING, 0x01)
+}
+
+def open() {
+    log.info "open()"
+    sendEvent(name: "windowShade", value: "opening")
+    zigbee.command(CLUSTER_WINDOW_COVERING, 0x00)
+}
+
+def setLevel(data, rate=null) {
+    log.info "setLevel()"
+
+    if(data == null){
+        data = 100
+    }
+    Integer currentLevel = state.level
```

Comment:
It doesn't appear that you set state.level anywhere. Instead of state.level, you can just use `device.currentValue("level")`. Keep in mind, `if (level > currentLevel)` will through an exception if currentLevel is null.